streams/readable-byte-streams/properties.js

```diff
@@ -0,0 +1,147 @@
+  assert_equals(rsReader.read.length, 1, 'read has 1 parameter');
+  assert_equals(rsReader.releaseLock.length, 0, 'releaseLock has no parameters');
+
+}, 'ReadableStreamBYOBReader instances should have the correct list of properties');
+
+promise_test(t => {
+
+  const rs = new ReadableStream({
+    pull(controller) {
+      return t.step(() => {
+        const byobRequest = controller.byobRequest;
+
+        const methods = ['constructor', 'respond', 'respondWithNewView'];
+        const properties = methods.concat(['view']).sort();
+
+        const proto = Object.getPrototypeOf(byobRequest);
+
+        assert_array_equals(Object.getOwnPropertyNames(proto).sort(), properties);
+
+        for (const m of methods) {
+          const propDesc = Object.getOwnPropertyDescriptor(proto, m);
+          assert_equals(propDesc.enumerable, false, 'method should be non-enumerable');
+          assert_equals(propDesc.configurable, true, 'method should be configurable');
+          assert_equals(propDesc.writable, true, 'method should be writable');
+          assert_equals(typeof byobRequest[m], 'function', 'should have a method');
+          const expectedName = m === 'constructor' ? 'ReadableStreamBYOBRequest' : m;
+          assert_equals(byobRequest[m].name, expectedName, 'method should have the correct name');
+        }
+
+        const viewPropDesc = Object.getOwnPropertyDescriptor(proto, 'view');
+        assert_equals(viewPropDesc.enumerable, false, 'view should be non-enumerable');
+        assert_equals(viewPropDesc.configurable, true, 'view should be configurable');
+        assert_not_equals(viewPropDesc.get, undefined, 'view should have a getter');
+        assert_equals(viewPropDesc.set, undefined, 'view should not have a setter');
+        assert_not_equals(byobRequest.view, undefined, 'has a non-undefined view property');
+        assert_equals(byobRequest.constructor.length, 2, 'constructor has 1 parameter');
+        assert_equals(byobRequest.respond.length, 1, 'respond has 1 parameter');
+        assert_equals(byobRequest.respondWithNewView.length, 1, 'releaseLock has 1 parameter');
+
+        byobRequest.respond(1);
+
+      });
+    },
+    type: 'bytes' });
+  const reader = rs.getReader({ mode: 'byob' });
+  return reader.read(new Uint8Array(1));
+
+}, 'ReadableStreamBYOBRequest instances should have the correct list of properties');
+
+test(() => {
+
+  const methods = ['close', 'constructor', 'enqueue', 'error'];
+  const accessors = ['byobRequest', 'desiredSize'];
+  const properties = methods.concat(accessors).sort();
+
+  let controller;
+  new ReadableStream({
+    start(c) {
+      controller = c;
+    },
+    type: 'bytes'
+  });
+  const proto = Object.getPrototypeOf(controller);
+
+  assert_array_equals(Object.getOwnPropertyNames(proto).sort(), properties);
+
+  for (const m of methods) {
+    const propDesc = Object.getOwnPropertyDescriptor(proto, m);
```

Comment:
Lots of duplication here and elsewhere, but meh...

Reply:
I know. I wanted to use the structure I used in [writable-streams/properties.js](https://github.com/w3c/web-platform-tests/blob/master/streams/writable-streams/properties.js) but the need to fetch ReadableStreamBYOBRequest asynchronously meant it wouldn't work directly. So then I just decided to go ahead and land this before the scope creep got any worse.